api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -120,6 +121,17 @@ public static Schema assignFreshIds(Schema schema, NextID nextId) {
         .fields());
   }
 
+  /**
+   * Assigns strictly increasing fresh ids for all fields in a schema, starting from 0.
```

Comment:
When a new ID is created, the IDs are assigned starting from 1. Can you update this to do the same to avoid confusion later on?

Reply:
Actually, I was able to fix this. I'll merge it.

Reply:
Thanks for the fix.